server/src/Packages/ExamplePackage/ExamplePackageProvider.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Gernzy\Server\Packages\ExamplePackage;
+
+use Gernzy\Server\Exceptions\GernzyException;
+use Gernzy\Server\Listeners\BeforeCheckout;
+use Gernzy\Server\Packages\ExamplePackage\Actions\ExampleBeforeCheckout;
+use Illuminate\Support\ServiceProvider;
+
+class ExamplePackageProvider extends ServiceProvider
+{
+    /**
+     * Register any application services.
+     *
+     * @return void
+     */
+    public function register()
+    {
+        $this->app->bind('foo', function ($app) {
+            // return new Bar();
+        });
+    }
+
+    /**
+     * Boot runs after register, and after all packages have been registered.
+     *
+     * @return void
+     */
+    public function boot()
+    {
+        // Example check if corresponding Event exists
+        $events = config('events');
+
+        // Check if config has values and the appropriate Event is present
+        if (isset($events) && !array_key_exists(BeforeCheckout::class, $events) && !class_exists(BeforeCheckout::class)) {
```

Comment:
@AlexeiaW this looks right, although I think we can make format this in a way that creates a bit of a convention. 

Within the service provider public vars, we could have

var $requiredActions = [
  BeforeCheckout::class
];

And then iterate through that. 

Possibly a better option here is also to extend ExamplePackageProvider from an already extended service provider. We could call that GernzyServiceProvider, which would provide the boot validation method we want like 

public function validateConfig(){
  foreach( $this->requiredActions as $action ){ /*... */ } 
}

Then in boot we could call this method.

public function boot(){
  $this->validateConfig();
}

Let me know what you think.